examples/cpp/seerep_ros_communication/src/rosbagDumper.cpp

```diff
@@ -0,0 +1,339 @@
+#include "seerep_ros_communication/rosbagDumper.h"
+
+namespace seerep_grpc_ros
+{
+RosbagDumper::RosbagDumper(std::string bagPath, std::string hdf5FilePath, std::string project_frame_id,
+                           std::string project_name, std::string topicImage, std::string topicCameraIntrinsics,
+                           std::string topicDetection, std::string detectionCategory, std::string topicTf,
+                           std::string topicTfStatic, std::string topicGeoAnchor, float distanceCameraGround)
+  : hdf5FilePath(hdf5FilePath)
+  , project_frame_id(project_frame_id)
+  , project_name(project_name)
+  , topicImage(topicImage)
+  , topicCameraIntrinsics(topicCameraIntrinsics)
+  , topicDetection(topicDetection)
+  , detectionCategory(detectionCategory)
+  , topicTf(topicTf)
+  , topicTfStatic(topicTfStatic)
+  , topicGeoAnchor(topicGeoAnchor)
+  , distanceCameraGround(distanceCameraGround)
+{
+  auto write_mtx = std::make_shared<std::mutex>();
+  std::shared_ptr<HighFive::File> hdf5_file =
+      std::make_shared<HighFive::File>(hdf5FilePath, HighFive::File::OpenOrCreate);
+  m_ioCoreGeneral = std::make_shared<seerep_hdf5_core::Hdf5CoreGeneral>(hdf5_file, write_mtx);
+  m_ioTf = std::make_shared<seerep_hdf5_fb::Hdf5FbTf>(hdf5_file, write_mtx);
+  m_ioImage = std::make_shared<seerep_hdf5_fb::Hdf5FbImage>(hdf5_file, write_mtx);
+  m_ioImageCore = std::make_shared<seerep_hdf5_core::Hdf5CoreImage>(hdf5_file, write_mtx);
+  m_ioPoint = std::make_shared<seerep_hdf5_fb::Hdf5FbPoint>(hdf5_file, write_mtx);
+
+  m_ioCoreGeneral->writeProjectname(std::filesystem::path(bagPath).filename());
+  m_ioCoreGeneral->writeProjectFrameId("map");
+
+  bag.open(bagPath);
+
+  getGeoAnchor();
+  getCameraIntrinsic();
+  iterateAndDumpTf();
+  iterateAndDumpImages();
+  iterateAndDumpDetections();
+}
+
+RosbagDumper::~RosbagDumper()
+{
+  bag.close();
+}
+
+void RosbagDumper::getGeoAnchor()
+{
+  for (rosbag::MessageInstance const m : rosbag::View(bag, rosbag::TopicQuery(topicGeoAnchor)))
```

Comment:
I would suggest using a reference for all loops. 
```suggestion
  for (rosbag::MessageInstance const& m : rosbag::View(bag, rosbag::TopicQuery(topicGeoAnchor)))
```